Robot Code/src/org/usfirst/frc/team3316/robot/subsystems/Chassis.java

```diff
@@ -240,4 +250,59 @@ public double getPitch()
 	{
 		return navx.getRoll();
 	}
+
+	public double getRoll()
+	{
+		return navx.getPitch();
+	}
+
+	public double getYaw()
+	{
+		return fixYaw(navx.getYaw());
+	}
+
+	// Returns the same heading in the range (-180) to (180)
+	private static double fixYaw(double heading)
+	{
+		double toReturn = heading % 360;
+
+		if (toReturn < -180)
+		{
+			toReturn += 360;
+		}
+		else if (toReturn > 180)
+		{
+			toReturn -= 360;
+		}
+		return toReturn;
+	}
+
+	public double getLeftSpeed()
+	{
+		return leftEncoder.getRate(); // Returns the speed in meter per
+										// second units.
+	}
+
+	public double getRightSpeed()
+	{
+		return rightEncoder.getRate(); // Returns the speed in meter per
+										// second units.
+	}
+
+	public double getDistance()
+	{
+		return (rightEncoder.getDistance() + leftEncoder.getDistance()) / 2;
+	}
+
+	public void resetEncoders()
+	{
```

Comment:
What do we need to reset the encoders for? Generally, it might cause more problems to other commands relying on the encoder output, and I think it is a safer practice that each command will calcaccuulate the difference it wants to reach from the current position, instead of resetting the accumulator.